src/ui/Combobox/ComboboxValues.ts

```diff
@@ -54,13 +54,26 @@ export class ComboboxValues implements IComboboxValues {
         { id: `${this.combobox.id}-value-${index}`, className: 'coveo-combobox-value', role: 'option', tabindex: 0 },
         value.element
       ).el;
+      this.highlightCurrentQueryInSearchResults(value.element);
       value.element = elementWrapper;
       fragment.appendChild(value.element);
     });
 
     this.element.appendChild(fragment);
   }
 
+  private highlightCurrentQueryInSearchResults(searchResult: HTMLElement) {
+    if (this.combobox.options.highlightValueClassName) {
+      const regex = new RegExp('(' + this.combobox.element.querySelector('input').value + ')', 'ig');
+      let result = $$(searchResult).hasClass(this.combobox.options.highlightValueClassName)
+        ? searchResult
+        : $$(searchResult).find(`.${this.combobox.options.highlightValueClassName}`);
```

Comment:
Hmm, I'm not clear on this ternary condition?
If a result has the class, we return the result, and if it does not, we search the children for it?
Is it possible that we do not find the class?

Reply:
Yes you're right, it should return in case the result if undefined

Reply:
I'm not sure if it's possible. If the search has value(s), then there should be a node with the class. If the search has no value, then this code isn't called, but I'm not 100% certain. I'll make it return if result is undefined to make it more robust.